docs/how-to/topics/api-design.rst

```diff
@@ -0,0 +1,50 @@
+==========
+API design
+==========
+
+Toga's API is structured around the following principles:
+
+Properties
+==========
+
+Toga prefers to expose an object's state using property notation (e.g.
+``widget.property``) rather than getter or setter methods.
```

Comment:
Should probably mention that properties should follow [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle) - if a widget stores text, any input should be coerced into a string when set, and be guaranteed to return text when retrieved.

Reply:
Also -uses as setter *wherever possible*. `WebView.set_content()` is a notable example, because it requires a URL as well as the HTML content.

Reply:
Done.